src/config/mod.rs

```diff
@@ -127,4 +175,42 @@ mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn test_config_comma_separated() -> eyre::Result<()> {
+        let config = "0:0,0:1,0:0-1,warboy(1)*1,warboy(2)*2,npu0pe0".parse::<DeviceConfig>()?;
```

Comment:
This is off-topic from this PR, and it just came to mind. I think that you already considered this case.

When multiple text representations are used, some items can be inclusive each other; e.g., `0:0, warboy(1)*1`. How can we handle this case?

I'd like to suggestion that device-api does't handle at all and the user-side of device-api should handle this issue. Do you agree with that? 

In furiosa-rt, it would be great if furiosa-rt returns a kind of duplicate devices error in order to minimize potential user mistakes.

Reply:
Though I don't think `0:0,warboy(1)*1` is inclusive, it is true that we have to have in mind about such cases. For example, "0:0,0:0" for the trivial case.

What I've intended is that constructing `DeviceConfig` does not return an error, but calling `find_devices` will generate an error naturally. In any case, it is not recommended to use a text repr with duplicate devices.

Incidentally as a side effect, the following may occur. `"0:0,warboy(1)*1"` would succeed, but for `"warboy(1)*1,0:0"`, `warboy(1)` may take `npu0pe0` so `find_devices` might fail. However, I won't handle this elegantly in device-api.

Reply:
Thank you for your comments. I understood its behavior. It would be great if you add more tests for that case.

Reply:
I added some more tests in [1542a71](https://github.com/furiosa-ai/device-api/pull/62/commits/1542a7125fbd3f594bd40822f58872f82ce9de81) and [53f6bae](https://github.com/furiosa-ai/device-api/pull/62/commits/53f6bae0ad866c951b18a1dee3eef958bee44923).